tests/blackbox/connectors/modbus/test_modbus_rpc.py

```diff
@@ -0,0 +1,447 @@
+from os import path
+import logging
+
+from pymodbus.exceptions import ConnectionException
+import pymodbus.client as ModbusClient
+from tb_rest_client.rest_client_ce import *
+from simplejson import load, loads
+
+from tests.base_test import BaseTest
+from tests.test_utils.gateway_device_util import GatewayDeviceUtil
+
+LOG = logging.getLogger("TEST")
+
+
+class ModbusRpcTest(BaseTest):
+    CONFIG_PATH = path.join(path.dirname(path.dirname(path.dirname(path.abspath(__file__)))),
+                            "data" + path.sep + "modbus" + path.sep)
+
+    client = None
+    gateway = None
+    device = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # super(ModbusRpcTest, cls).setUpClass()
+
+        # ThingsBoard REST API URL
+        url = GatewayDeviceUtil.DEFAULT_URL
+
+        # Default Tenant Administrator credentials
+        username = GatewayDeviceUtil.DEFAULT_USERNAME
+        password = GatewayDeviceUtil.DEFAULT_PASSWORD
+
+        with RestClientCE(url) as cls.client:
+            cls.client.login(username, password)
+
+            cls.gateway = cls.client.get_tenant_devices(10, 0, text_search='Gateway').data[0]
+            assert cls.gateway is not None
+
+            while not cls.is_gateway_connected():
+                LOG.info('Gateway connecting to TB...')
+                sleep(1)
+
+            LOG.info('Gateway connected to TB')
+
+            cls.device = cls.client.get_tenant_devices(10, 0, text_search='Temp Sensor').data[0]
+            assert cls.device is not None
+
+    @classmethod
+    def tearDownClass(cls):
+        super(ModbusRpcTest, cls).tearDownClass()
+
+        client = ModbusClient.ModbusTcpClient('modbus-server', port=5021)
+        client.connect()
+        try:
+            client.write_register(28, 10, 1)
+        except ConnectionException:
+            # will call pymodbus.exceptions.ConnectionException because of restarting the server
+            pass
+
+        client.close()
+        sleep(2)
+
+    @classmethod
+    def is_gateway_connected(cls):
```

Comment:
It will be better to check the server attribute "lastConnectedTime" with "active" attribute, because active attribute is not a signal that the device is connected.